orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -305,13 +306,15 @@ static Status ModifyParametersForOptimizerPartitioning(
         int64_t size_for_current_rank = offset + tensor_count - rank_start;
         std::vector<TensorShape> view_shapes = {{size_for_previous_rank}, {size_for_current_rank}, {0}};
         std::vector<bool> enabled = {false, true};
+        weight_partition_info[weight_argdef.name].original_dimension = tensor_shape.GetDims();
```

Comment:
Should original_dimension set for 
// 1. parameter is fully handled by this rank 
And AddParameterPartition is not called?
I can change the if branch a little bit so that original_dimension is not set for case 1.

Reply:
Synced offline. Changed if branch a bit and only have one instance of `weight_partition_info[weight_argdef.name].original_dimension = tensor_shape.GetDims();` now